examples/miscellaneous/miscellaneous_ex15/miscellaneous_ex15.C

```diff
@@ -0,0 +1,301 @@
+      FEBase * cfe = libmesh_nullptr;
+
+      if (elem->infinite())
+        cfe = inf_fe.get();
+      else
+        cfe = fe.get();
+
+      // The element Jacobian * quadrature weight at each integration point.
+      const std::vector<Real>&                        JxW = cfe->get_JxWxdecay_sq();
+      // The element shape functions evaluated at the quadrature points.
+      const std::vector<std::vector<Real> >&         phi  = cfe->get_phi_over_decayxR();
+      const std::vector<std::vector<RealGradient> >& dphi = cfe->get_dphi_over_decayxR();
+      const std::vector<Real>&                      sob_w = cfe->get_Sobolev_weightxR_sq();
+      const std::vector<RealGradient>&             dsob_w = cfe->get_Sobolev_dweightxR_sq();
+
+      // Compute the element-specific data for the current
+      // element. This involves computing the location of the
+      // quadrature points (q_point) and the shape functions
+      // (phi, dphi) for the current element.
+      cfe->reinit (elem);
+      M.resize (dof_indices.size(), dof_indices.size());
+      f.resize (dof_indices.size());
+
+      Real rho;
+      // check if charged_objects contains elem->id(), we add the corresponding charge to the rhs vector
+      if (std::find(charged_objects.begin(), charged_objects.end(), elem->id()) != charged_objects.end())
+        rho=1./elem->volume();
+      else
+        rho = 0;
+
+      const unsigned int max_qp = cfe->n_quadrature_points();
+      for (unsigned int qp=0; qp<max_qp; ++qp)
+        {
+          const unsigned int n_sf = cfe->n_shape_functions();
+          for (unsigned int i=0; i<n_sf; ++i)
+            {
+              if (rho > 0)
+                f(i) -=4*pi*JxW[qp]*rho*phi[i][qp];
+
+              // store the test functions derivative (which contains the extra sobolev weight)
+              // separately; so we don't need to recompute it for each j.
+              const RealGradient dphi_i=dphi[i][qp]*sob_w[qp]+phi[i][qp]*dsob_w[qp];
+              for (unsigned int j=0; j<n_sf; ++j)
+                {
+                  M(i,j) += JxW[qp]*dphi_i*dphi[j][qp];
+                }
+            }
+        }
+      dof_map.constrain_element_matrix_and_vector(M, f, dof_indices);
+      f_system.matrix->add_matrix(M, dof_indices);
+      f_system.rhs->add_vector(f, dof_indices);
+    }
+  f_system.rhs->close();
+  f_system.matrix->close();
+  /**
+   * All done!
+   */
+  return;
+}
+
+int main (int argc, char** argv)
+{
+  // Initialize libMesh and the dependent libraries.
+  LibMeshInit init (argc, argv);
+
+  // Tell the user what we are doing.
+  std::cout << "Running " << argv[0];
+  for (int i=1; i<argc; i++)
+    std::cout << " " << argv[i];
+  std::cout << std::endl << std::endl;
+
+#ifndef LIBMESH_ENABLE_INFINITE_ELEMENTS
+  libmesh_example_requires(false, "--enable-ifem");
+#else
+
+#ifdef LIBMESH_ENABLE_SECOND_DERIVATIVES
+  // UniformRefinement uses second derivatives if they are enabled,
```

Comment:
Oh, well that's a bug on my part.  Looks like I'm requesting the kitchen sink, then deciding after the fact which parts of it actually need to be used?  It'll be easy to fix that.  I'll put the fix in a separate PR after this is merged.